pkg/verification/verification_service_test.go

```diff
@@ -150,6 +158,211 @@ func (s *TestVerificationServiceSuite) TestSendVerifyMessageFails() {
 	require.Empty(s.T(), userSignup.Annotations[v1alpha1.UserSignupVerificationCodeAnnotationKey])
 }
 
+func (s *TestVerificationServiceSuite) TestVerifyCode() {
+	// given
+	rr := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(rr)
+	svc, _ := s.createVerificationService()
+	now := time.Now()
+
+	s.T().Run("verification ok", func(t *testing.T) {
+
+		userSignup := &v1alpha1.UserSignup{
+			TypeMeta: v1.TypeMeta{},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "123",
+				Namespace: "test",
+				Annotations: map[string]string{
+					v1alpha1.UserSignupUserEmailAnnotationKey:        "sbryzak@redhat.com",
+					v1alpha1.UserVerificationAttemptsAnnotationKey:   "0",
+					v1alpha1.UserSignupVerificationCodeAnnotationKey: "123456",
+					v1alpha1.UserVerificationExpiryAnnotationKey:     now.Add(10 * time.Second).Format(verification.TimestampLayout),
+				},
+				Labels: map[string]string{
+					v1alpha1.UserSignupPhoneNumberLabelKey: "+1NUMBER",
```

Comment:
did we update anything to add this phone number field when a usersignup is created? what happens if phone verification is required and no phone number exists? just a thought after testing on a temp cluster. the usersignup is created for me without a phonenumber, hence sending a verification fails unless i patch my usersignup to contain my phone number..

Reply:
The user will provide their phone number when they initiate phone verification.  It won't be present in the UserSignup until then.

Reply:
Yeah.. we will need to cover the entire scenario in our e2e tests when everything is ready.